app/lib/analyzer/task_sources.dart

```diff
@@ -4,24 +4,93 @@
 
 import 'dart:async';
 
+import 'package:gcloud/db.dart';
+
+import '../frontend/models.dart';
 import '../shared/task_scheduler.dart';
 
+import 'models.dart';
+
+/// Creates a task when a version uploaded in the past 10 minutes has no
+/// analysis yet.
 class DatastoreHeadTaskSource extends PollingTaskSource {
-  DatastoreHeadTaskSource() : super(new Duration(minutes: 10));
+  final DatastoreDB _db;
+  DateTime _lastTs;
+  DatastoreHeadTaskSource(this._db) : super(const Duration(minutes: 1));
 
   @override
-  Stream<Task> pollTasks() {
-    // TODO: implement pollTasks
-    return new Stream.empty();
+  Future poll() async {
+    final DateTime now = new DateTime.now().toUtc();
+    final DateTime tenMinutesAgo = now.subtract(const Duration(minutes: 10));
+    DateTime minCreated;
+    if (_lastTs == null) {
+      minCreated = tenMinutesAgo;
+    } else if (_lastTs.isBefore(tenMinutesAgo)) {
+      // more than ten minutes passed since the last poll
+      minCreated = _lastTs;
+    } else {
+      minCreated = tenMinutesAgo;
+    }
+    _lastTs = tenMinutesAgo;
```

Comment:
Since you unconditionally set `_lastTs = tenMinutesAgo`, the `else if (_lastTs.isBefore(tenMinutesAgo)` will always be true (except in the very first call), so I don't think we ever go into the `else` branch.

Also as mentioned in another comment, `_lastTs` can be simply set to the newest `PackageVersion.created` that the query returned.